pkg/dns/azure/client/auth.go

```diff
@@ -116,6 +120,15 @@ func getAuthorizerForResource(config Config) (autorest.Authorizer, error) {
 		if err != nil {
 			return nil, err
 		}
+	} else if userAssignedIdentityCredentialsFilePath != "" {
+		clientOptions := azcore.ClientOptions{
```

Comment:
Is there a reason not to call it `options`, consistent with the other cases?  

It's a tiny detail, but the inconsistency jumped out at me.  

On an unrelated note, this `if`/`else if`/`else if`/`else` chain has grown to the point that it should be a `switch` block now.  That refactoring can be left to a separate commit to avoid complicating this PR.